basic-steps/src/main/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepExecution.java

```diff
@@ -15,6 +20,8 @@
     private BodyExecution body;
     private transient ScheduledFuture<?> killer;
 
+    private long end;
```

Comment:
You need to at least consider what happens to an existing build that was inside `timeout` when the upgrade happens: this field will be initialized to `0`. Now due to #161, fixed in this same release (assuming you get this review done soon), such a build would have ignored the timeout. Looks like the behavior now will be that the build gets canceled even if it was within its time (since `0 > now` will be false). Not a big deal—a one-time-only effect—but worth considering.